### DISCUSSION TITLE
More Insert FX

### DISCUSSION THREAD

**Discussion by Maxklui** [bangcorrupt/hacktribe#75] [Ideas]
Hello @bangcorrupt 

I had an idea for an Update. Is it possible to add more IFX effects to the Electribe. 
Or is it possible to add an Sidechain compression to the Electribe. I mean not the 
Pumpereffect

Thank you very much, and I hope it comes to the Electribe

Greetings Max

**Comment by bangcorrupt** [comment]
More FX presets and an external FX preset editor will definitely happen. Adding an extra FX unit to the chain likely won't happen.  I doubt I can improve on the DSP efficiency compared to engineers with source code.  

I'd like to modify the FX buss structure to have multiple MFX send groups, perhaps moving the expensive FX like delay to group busses.  The FX structure of the EMX worked well, it would be nice to combine that with the individual channel inserts of e2.  Reversing the Mini Kaoss Pad 2 and KingKorg firmwares could help with this, but it's a big change and probably won't happen.    

I don't know if the existing FX engine allows for a sidechain input to any of the compressors.  I'll have a debug adapter for the DSP next month and will be able to find out more.

**Reply by Maxklui** [reply]
Very, very cool. I'm definitely looking forward to the new effects. Effect chains will also be possible, or not?

**Reply by bangcorrupt** [reply]
No FX chains.

**Reply by Maxklui** [reply]
@bangcorrupt What else will be added to the Hacktribe firmware. Would I be really interested?I mean in general? 😃😃

**Reply by bangcorrupt** [reply]
Look for anything labeled enhancement in [discussions](https://github.com/bangcorrupt/hacktribe/discussions?discussions_q=label%3Aenhancement) and [issues](https://github.com/bangcorrupt/hacktribe/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) There's no guarantee any of these will ever work.  Assume nothing will be implemented, until it is.

**Reply by Maxklui** [reply]
@bangcorrupt I apologize for the many questions. Is there currently what I can still install for the firmware?

**Reply by bangcorrupt** [reply]
> Is there currently what I can still install for the firmware?

I'm sorry I don't understand this question.

**Reply by Maxklui** [reply]
@bangcorrupt I mean Features like the groov type. Than the new groove types are not working for me?

**Reply by bangcorrupt** [reply]
> the new groove types are not working for me

The new groove templates should show in the menu like the rest.  The script [e2_groove_template.py](https://github.com/bangcorrupt/e2-scripts/blob/fb1e845c7d75734d1374f8e2b9fea6d09ee49c5a/e2_groove_template.py) demonstrates how to edit groove templates.  At the moment these are only stored in RAM, so edits will be lost after a power cycle.  Writing to the flash will be added later.

**Reply by maxtribe** [reply]
@bangcorrupt how can I install this template

**Reply by bangcorrupt** [reply]
That's a python script for editing groove templates.  Later there will be a GUI editor.

**Comment by maks** [comment]
Wow having even just a second Mfx bus for a group would be amazing! Just that would allow having 1 Mfx for a group of channels for an expensive fx (reverb of course) and then having everything going to a 2nd Mfx that could be used for performance, eg. applying DJ style HP-LP filter using the x-y pad.

**Reply by bangcorrupt** [reply]
I'm hoping I'll be able to see a clear buss structure is the DSP code.  I ordered the debug adapter, so we should know more in a few months.
